src/parse/grammar/checks.ts

```diff
@@ -427,6 +430,53 @@ export function validateAmbiguousAlternationAlternatives(topLevelRule:gast.Rule,
     return errors
 }
 
+export class RepitionCollector extends gast.GAstVisitor {
+    public allProductions:gast.IProduction[] = []
+
+    public visitRepetitionWithSeparator(manySep:gast.RepetitionWithSeparator):void {
+        this.allProductions.push(manySep)
+    }
+
+    public visitRepetitionMandatory(atLeastOne:gast.RepetitionMandatory):void {
+        this.allProductions.push(atLeastOne)
+    }
+
+    public visitRepetitionMandatoryWithSeparator(atLeastOneSep:gast.RepetitionMandatoryWithSeparator):void {
+        this.allProductions.push(atLeastOneSep)
+    }
+
+    public visitRepetition(many:gast.Repetition):void {
+        this.allProductions.push(many)
+    }
+}
+
+export function validateSomeNonEmptyLookaheadPath(topLevelRules:gast.Rule[], maxLookahead:number):IParserDefinitionError[] {
+    let errors = []
+    forEach(topLevelRules, (currTopRule) => {
+        let collectorVisitor = new RepitionCollector()
+        currTopRule.accept(collectorVisitor)
+        let allRuleProductions = collectorVisitor.allProductions
+        forEach(allRuleProductions, (currProd) => {
+            let prodType = getProdType(currProd)
+            let currOccurrence = currProd.occurrenceInParent
+            let paths = getLookaheadPathsForOptionalProd(currOccurrence, currTopRule, prodType, maxLookahead)
+            let pathsInsideProduction = paths[0]
+            if (isEmpty(flatten(pathsInsideProduction))) {
+                let dslName = getProductionDslName(currProd)
+                let errMsg = `Empty lookahead paths found in <${dslName}${currOccurrence}> within Rule <${currTopRule.name}>.\n` +
```

Comment:
By the way, why not add a getter property on `IProductionWithOccurrence` that returns the original mention as-is by making use of `implicitOccurrenceIndex`? Using the flag externally would require us to assemble this information yourself in a bunch of places.